File: CS-Examples/07_Conversion/ToSVG.cs
using System;
using System.Windows.Forms;
using Spire.Xls;
using System.IO;

namespace ToSVG
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void btnRun_Click(object sender, System.EventArgs e)
        {
            // Create a workbook
            Workbook workbook = new Workbook();

            // Load a file from the specified path into the workbook
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\ToSVG.xlsx");

            // Iterate through each worksheet in the workbook
            for (int i = 0; i < workbook.Worksheets.Count; i++)
            {
                // Create a FileStream to write the SVG content to a file
                FileStream fs = new FileStream(string.Format("sheet{0}.svg", i), FileMode.Create);
                // Convert the worksheet to SVG and write it to the FileStream
                workbook.Worksheets[i].ToSVGStream(fs, 0, 0, 0, 0);
                // Flush and close the FileStream to ensure data is written and resources are released
                fs.Flush();
                fs.Close();
            }

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            // Launch the document
            FileViewer("sheet0.svg");
            
        }
        private void FileViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
    }
}

File: CS-Examples/07_Conversion/ToSVG.vb
Imports System.Data.OleDb
Imports System.Collections
Imports System.ComponentModel

Imports Spire.Xls
Imports System.IO

Namespace ToSVG
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new Workbook object to represent an Excel workbook
            Dim workbook As New Workbook()

            ' Load an existing Excel file from the specified path
            workbook.LoadFromFile("..\..\..\..\..\..\Data\ToSVG.xlsx")

            ' Iterate through each worksheet in the workbook
            For i As Integer = 0 To workbook.Worksheets.Count - 1

                ' Create a FileStream object with the filename for the SVG file
                Dim fs As New FileStream(String.Format("sheet{0}.svg", i), FileMode.Create)

                ' Convert the current worksheet to SVG format and save it to the FileStream
                workbook.Worksheets(i).ToSVGStream(fs, 0, 0, 0, 0)

                ' Flush the FileStream to ensure all data is written
                fs.Flush()

                ' Close the FileStream to release resources
                fs.Close()
            Next i
            ' Release the resources used by the workbook
            workbook.Dispose()

            Process.Start("sheet0.svg")
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class
End Namespace
